dll/ntdll/ldr/ldrapi.c

```diff
@@ -1476,10 +1476,18 @@ LdrUnloadDll(IN PVOID BaseAddress)
                                                    LdrEntry->EntryPointActivationContext);
 
             /* Call the entrypoint */
-            LdrpCallInitRoutine(LdrEntry->EntryPoint,
-                                LdrEntry->DllBase,
-                                DLL_PROCESS_DETACH,
-                                NULL);
+            _SEH2_TRY
+            {
+                LdrpCallInitRoutine(LdrEntry->EntryPoint,
```

Comment:
Any chance it's supposed to be LdrpCallInitRoutine that does the SEH?

Reply:
That is also possible, guess it is time to write a test...

Reply:
Based on the SEH frame depth difference of only 1 in 2k3 when {{DLL_PROCESS_DETACH}} is called, i'd go out on a limb and say that {{LdrpCallInitRoutine}} does not do this.
Later versions show that both the {{ATTACH}} and {{DETACH}} have a higher seh frame count, so they probably added it later.